src/org/opendatakit/briefcase/export/SubmissionParser.java

```diff
@@ -0,0 +1,187 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.opendatakit.briefcase.export;
+
+import static org.apache.commons.codec.binary.Base64.decodeBase64;
+import static org.opendatakit.briefcase.export.CipherFactory.signatureDecrypter;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.createTempDirectory;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.list;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.newInputStream;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.stripFileExtension;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.io.UnsupportedEncodingException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.IllegalBlockSizeException;
+import org.kxml2.io.KXmlParser;
+import org.kxml2.kdom.Document;
+import org.opendatakit.briefcase.model.CryptoException;
+import org.opendatakit.briefcase.reused.OptionalProduct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+/**
+ * This class holds the main submission parsing code.
+ */
+class SubmissionParser {
+  private static final Logger log = LoggerFactory.getLogger(SubmissionParser.class);
+
+  /**
+   * Parses all submission files in a form's directory and returns a {@link List} of {@link Submission} instances.
+   *
+   * @param formDir     the {@link Path} directory of the form
+   * @param isEncrypted a {@link Boolean} indicating if the form is encrypted or not
+   * @param privateKey  {@link PrivateKey} instance wrapped inside an {@link Optional} instance
+   * @param dateRange   a {@link DateRange} to filter submissions that are contained in it
+   * @return a {@link List} of {@link Submission} instances
+   */
+  static Stream<Submission> parseAllInFormDir(Path formDir, boolean isEncrypted, Optional<PrivateKey> privateKey, DateRange dateRange) {
+    Path instancesDir = formDir.resolve("instances");
+    if (!Files.exists(instancesDir) || !Files.isReadable(instancesDir))
+      return Stream.empty();
+
+    return list(instancesDir).parallel()
+        // Normally, we would use the Streams API to filter out files that
+        // are not contained in the given date range, but parsing is expensive
+        // enough to make it all in one pass and use Optionals to account
+        // for that
+        .map(path -> parseSubmission(path.resolve("submission.xml"), dateRange, isEncrypted, privateKey))
```

Comment:
I had an error here:

java.io.UncheckedIOException: java.nio.file.FileSystemException: /private/tmp/ODK Briefcase Storage/forms/Independent Monitoring/instances/.DS_Store/submission.xml: Not a directory

Here is one solution:
```
        // for that
        .filter(path -> !path.toFile().isHidden())
        .map(path -> parseSubmission(path.resolve("submission.xml"), dateRange, isEncrypted, privateKey))
```

Reply:
Something doesn't seem right... Could you copy the full trace here, please?

I don't understand how you're getting that error there and what has to do with `.DS_Store` beind hidden...

Reply:
Browse your data with Finder (are you using MacOS?) and it will create the .DS_Store directories.